trustpinning/trustpin_test.go

```diff
@@ -0,0 +1,33 @@
+package trustpinning
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestWildcardMatch(t *testing.T) {
+	testCerts := map[string][]string{
+		"docker.io/library/ubuntu": {"abc"},
+		"docker.io/endophage/b*":   {"def"},
+		"docker.io/endophage/*":    {"xyz"},
+	}
+
+	// wildcardMatch should ONLY match wildcarded names even if a specific
+	// match is present
+	res, ok := wildcardMatch("docker.io/library/ubuntu", testCerts)
+	require.Nil(t, res)
+	require.False(t, ok)
+
+	// wildcard match should match on segment boundaries
+	res, ok = wildcardMatch("docker.io/endophage/foo", testCerts)
+	require.Len(t, res, 1)
+	require.Equal(t, "xyz", res[0])
+	require.True(t, ok)
+
+	// wildcardMatch should also match between segment boundaries, and take
+	// the first match is finds as the ONLY match.
```

Comment:
ah sorry, I need to update the comment. I originally did first and was just going to have the user prioritize in their config file however they want, but that was before I remembered the random map ordering. Will update the comment.